libraries/effect/src/main/java/androidx/media3/effect/VideoCompositor.java

```diff
@@ -48,8 +48,11 @@ interface Listener {
   /**
    * Registers a new input source, and returns a unique {@code inputId} corresponding to this
    * source, to be used in {@link #queueInputTexture}.
+   *
+   * @param sequenceIndex The sequence index of the input source, which is can be used to determine
+   *     the order of the input sources.
    */
-  int registerInputSource();
+  int registerInputSource(int sequenceIndex);
```

Comment:
a high level comment: it's strange that this method takes in an ID and returns another ID, where those IDs can potentially be the same. In the actual impl(in `DefaultVideoCompositor`) those two indices match and that's what fixes the z-order problem. 

I feel the approach could be a bit top down, that this method just takes a `sequenceIndex` but doesn't return any index. The passed-in index ultimately defines the z-order, wdyt? @dway123 please also take a look and see if you agree.

Reply:
@claincly This approach sounds good to me!

Let me change the code according to this design.

Reply:
@dway123 @claincly Implemented the top down approach for id passing as discussed.

Reply:
@claincly ptal?